packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -7071,3 +7071,52 @@ class _StatefulBuilderState extends State<StatefulBuilder> {
   @override
   Widget build(BuildContext context) => widget.builder(context, setState);
 }
+
+/// A widget that paints its area with a specified [Color] and then draws its
+/// child.
+class ColoredBox extends SingleChildRenderObjectWidget {
+  /// Creates a widget that paints its area with the specified [Color].
+  ///
+  /// The `color` parameter must not be null.
+  const ColoredBox({@required this.color, Widget child, Key key})
+      : assert(color != null),
+        super(key: key, child: child);
+
+  /// The color to paint the background area with.
+  final Color color;
+
+  @override
+  RenderColoredBox createRenderObject(BuildContext context) {
+    return RenderColoredBox(color: color);
+  }
+}
+
+/// A render object that paints its area with the specified [Color].
+class RenderColoredBox extends RenderProxyBox {
+  /// Creates a render object that paints its area with the specified [Color].
+  RenderColoredBox({@required Color color}) : _color = color;
+
+  /// The fill color for this render object.
+  ///
+  /// This parameter must not be null.
+  Color get color => _color;
+  Color _color;
+  set color(Color value) {
+    assert(value != null);
+    if (value == _color) {
+      return;
+    }
+    _color = value;
+    markNeedsPaint();
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (size != Size.zero) {
```

Comment:
I suppose to cover the corner of this corner case: `(size > Size.zero)`

Reply:
Done